test/runner/e2e_delete_rel_test.cpp

```diff
@@ -0,0 +1,198 @@
+    auto query = "MATCH (a:person)-[e:knows]->(b:animal) RETURN e.length";
+    // This check is to validate whether the query used in "deleteRelsAndQueryBWDList"
+    // scans the backward list of the knows table. If this test fails (eg. we have updated the
+    // planner), we should consider changing to a new query which scans the BWD list with
+    // the new planner.
+    validateQueryBestPlanJoinOrder(query, "HJ(a){E(a)S(b)}{S(a)}");
+    conn->beginWriteTransaction();
+    // We insert 10 rels: person[0..10]->animal0.
+    for (auto i = 0u; i < 10; i++) {
+        ASSERT_TRUE(
+            conn->query(getInsertKnowsRelQuery("person", "animal", i, 0 /* dstID */))->isSuccess());
+    }
+    // Then we delete rels: person[2,4,6,8]->animal0.
+    ASSERT_TRUE(
+        conn->query(getDeleteKnowsRelQuery("person", "animal", 2, 0 /* dstID */))->isSuccess());
+    ASSERT_TRUE(
+        conn->query(getDeleteKnowsRelQuery("person", "animal", 4, 0 /* dstID */))->isSuccess());
+    ASSERT_TRUE(
+        conn->query(getDeleteKnowsRelQuery("person", "animal", 6, 0 /* dstID */))->isSuccess());
+    ASSERT_TRUE(
+        conn->query(getDeleteKnowsRelQuery("person", "animal", 8, 0 /* dstID */))->isSuccess());
+    vector<string> expectedResult = {"0", "1", "3", "5", "7", "9"};
+    ASSERT_EQ(TestHelper::convertResultToString(*conn->query(query), true /* checkOutputOrder */),
+        expectedResult);
+}
+
+TEST_F(DeleteRelTest, MixedDeleteAndCreateRels) {
+    conn->beginWriteTransaction();
+    // We firstly delete rel person7|8 -> person0. Then insert person7->person0 and delete it again.
+    auto knowsRelQuery = "MATCH (a:animal)-[e:knows]->(:person) where a.ID <= 10 return e.length";
+    ASSERT_TRUE(
+        conn->query(getDeleteKnowsRelQuery("animal", "person", 8 /* srcID */, 0 /* dstID */))
+            ->isSuccess());
+    ASSERT_TRUE(
+        conn->query(getDeleteKnowsRelQuery("animal", "person", 7 /* srcID */, 0 /* dstID */))
+            ->isSuccess());
+    vector<string> expectedResult = {"0", "1", "2", "3", "4", "5", "6", "9", "10"};
+    ASSERT_EQ(
+        TestHelper::convertResultToString(*conn->query(knowsRelQuery), true /* checkOutputOrder */),
+        expectedResult);
+    ASSERT_TRUE(
+        conn->query(getInsertKnowsRelQuery("animal", "person", 7 /* srcID */, 0 /* dstID */))
+            ->isSuccess());
+    expectedResult = {"0", "1", "2", "3", "4", "5", "6", "7", "9", "10"};
+    ASSERT_EQ(
+        TestHelper::convertResultToString(*conn->query(knowsRelQuery), true /* checkOutputOrder */),
+        expectedResult);
+    ASSERT_TRUE(
+        conn->query(getDeleteKnowsRelQuery("animal", "person", 7 /* srcID */, 0 /* dstID */))
+            ->isSuccess());
+    expectedResult = {"0", "1", "2", "3", "4", "5", "6", "9", "10"};
+    ASSERT_EQ(
+        TestHelper::convertResultToString(*conn->query(knowsRelQuery), true /* checkOutputOrder */),
+        expectedResult);
+}
+
+TEST_F(DeleteRelTest, DeleteAllInsertedRels) {
```

Comment:
As we discussed, replace the names of this test and the DeleteMultipleRels below.

Reply:
Swapped the names for these two tests.